Fix certificate sass assets collection for comprehensive themes

angonz opened the pull request:
This change includes the directory where sass files for certificates are located within each comprehensive theme directory: lms/static/certificates/sass. This directory is used by the `paver update_assets` command.

edx-webhook commented:
Thanks for the pull request, @angonz! I've created [OSPR-5046](https://openedx.atlassian.net/browse/OSPR-5046) to keep track of it in JIRA, where we prioritize reviews. Please note that it may take us up to several weeks or months to complete a review and merge your PR.

Feel free to add as much of the following information to the ticket:
- supporting documentation
- [Open edX discussion forum threads](https://discuss.openedx.org/)
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).


Please let us know once your PR is ready for our review and all tests are green.




<!-- comment:external_pr -->
<!-- jenkins ok to test -->
<!-- data: eyJkcmFmdCI6IGZhbHNlfQ== -->

natabene commented:
@angonz Thank you for your contribution. Please let me know once it is ready for our review.

regisb commented:
@angonz I'm not a core committer here, but this looks good to me! Could you please just modify the commit title and description? Generally speaking, you should avoid writing *what* you are doing in the commit title, but *why* you are doing it. For instance, here a more appropriate commit title would be "Fix certificate static asset collection", or something similar. You should then add more words in the commit description about what you are doing precisely and the exact problem that you are solving. (Reference: https://chris.beams.io/posts/git-commit/)

angonz commented:
> @angonz Thank you for your contribution. Please let me know once it is ready for our review.

Thanks @regisb for your tips! I've changed the title and description (tell me if it looks better now), and added your tips to the contributor's guide.

Hi @natabene, yes it is ready for review.

regisb commented:
> I've changed the title and description (tell me if it looks better now), and added your tips to the contributor's guide.

Double win ;) thanks for that @angonz! Now you should just wait for a reviewer to come here and check out your PR. If no one shows up after 1-2 weeks you can come back here and post a comment. You should make sure that your PR gets merged in time for the Koa code freeze (Nov 9).

felipemontoya commented on pavelib/assets.py:
```diff
@@ -194,6 +194,17 @@ def get_theme_sass_dirs(system, theme_dir):
         })
 
+        # now compile theme sass files for certificate
+        if system == 'lms':
```
Is it necessary to make it specific to the lms. I thought certificates are also rendered by Studio in preview mode. I could be wrong there.

angonz replied:
In my tests, only by setting the theme for the lms works also for preview. I guess the preview engine uses the lms rendering routines.

felipemontoya commented on pavelib/assets.py:
```diff
@@ -194,6 +194,17 @@ def get_theme_sass_dirs(system, theme_dir):
+        # now compile theme sass files for certificate
+        if system == 'lms':
+            dirs.append({
+                "sass_source_dir": theme_dir / system / "static" / "certificates" / "sass",
```
Perpahs you would like to extract the dir in the same way it is done in lines 167-169 to a variable `certs_sass_dir` or similar.

angonz replied:
Good suggestion. It will improve readability.

felipemontoya reviewed (commented):
I left minor comments inline in the code. That is for the form.

Now for the , what I would like to understand is the consequences of adding the `certificates` dir to every theme. would it work if I don't add any new directory to an already existing theme?

And also if we are going to open the door to having more in the `static` dir than just the `sass` dir. Are there more of those?

I think that maybe @amalbas or @abutterworth can chime in on that.

natabene commented:
@felipemontoya Would you like to review this as a core committer?

felipemontoya commented:
I thought that my previous review was in the status of core committer. Should I do something differently?

natabene commented:
@felipemontoya Thanks for taking this PR. If you could in the future add "cc-reviewer" label to the PRs you want to review and merge that would be great. Please see more details here: https://openedx.atlassian.net/wiki/spaces/COMM/pages/1729200289/OSPR+Reviews+by+Core+Committers

angonz commented:
Hi @felipemontoya, is there anything on my side to do here? What should be the next step?

felipemontoya commented:
Hi @angonz, I was waiting for confirmation by Aura or Adam. However I asked @amalbas directly and she agrees on both having the `certificates` directory and the fact that only the lms needs to produce the styles.

Have you tried compiling a theme that does not have the `certificates` dir?

angonz commented:
Hi @felipemontoya, yes I have some themes with certificates dir and some others without, and it works perfectly well.

felipemontoya commented:
@angonz then this is good for me.
@natabene should I wait for a review by @abutterworth or just wait for decent hours and merge?

abutterworth reviewed (approved):
Sorry, for my tardiness. Looking at the changes this all looks good to me. (cc @davidjoy)

edx-webhook commented:
@angonz 🎉 Your pull request was merged!

Please take a moment to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=edx/edx-platform&entry.867055334=https://github.com/edx/edx-platform/pull/25322&entry.1484655318=https://github.com/angonz&entry.752974735=2020-10-09+20:42&entry.1917517419=2020-11-10+21:41&entry.2133058324=Yes) so we can improve your experience in the future.

felipemontoya commented:
@angonz thanks a lot for your contribution!
